src/views/LoginSuccess.vue

```diff
@@ -30,226 +29,108 @@ import { Component, Emit, Vue } from 'vue-property-decorator';
-     * number) which is incremented each time the Map is updated. The DOM is then fed not by the
-     * Map itself, but by a computed property (`this.accountsArray` in this case), which uses
-     * both the non-reactive Map and the reactive partner-attribute. The partner-attribute change
-     * triggers the recomputation, which in turn updates the DOM.
+     * Used to invalidate the cached value for getter `accountsArray`.
      */
-    private accounts: Map<string, AccountInfo> = new Map();
-    private accountsUpdateCount: number = 0;
+    private updateCount: number = 0;
 
-    private lastDerivedIndex: number = 0;
-    private lastActiveIndex: number = 0;
-
-    private result?: LoginResult;
-
-    private contentReady: boolean = false;
+    private retrievalFailed: boolean = false;
     private retrievalComplete: boolean = false;
 
     private async mounted() {
-        const currentWalletInfo = await WalletStore.Instance.get(this.keyguardResult.keyId);
-
-        // if there is a WalletInfo for the returned keyId already, use it
-        if (currentWalletInfo && currentWalletInfo.label) {
-            this.walletLabel = currentWalletInfo.label;
-            if (currentWalletInfo.accounts) {
-                this.accounts = currentWalletInfo.accounts;
-            }
-        }
-
         // The Keyguard always returns (at least) one derived Address,
-        // thus we can already create a complete WalletInfo object that
-        // can be displayed while waiting for the network.
-        // Before adding the WalletInfo object, make sure it is not set already.
-        this.keyguardResult.addresses.forEach((addressObj) => {
-            const address = new Nimiq.Address(addressObj.address);
-            if (!this.accounts.has(address.toUserFriendlyAddress())) {
-                const addressInfo = new AccountInfo(
-                    addressObj.keyPath,
-                    this.defaultAccountLabel,
-                    address,
+        const keyguardResultAccounts = this.keyguardResult.addresses.map((addressObj) => ({
+            address: new Nimiq.Address(addressObj.address).toUserFriendlyAddress(),
+            path: addressObj.keyPath,
+        }));
+
+        let tryCount = 0;
+        while (true) {
+            try {
+                ++tryCount;
+                const walletInfo = await WalletInfoCollector.collectWalletInfo(
+                    this.keyguardResult.keyType,
+                    this.keyguardResult.keyId,
+                    keyguardResultAccounts,
+                    (updatedWalletInfo) => this._onWalletInfoUpdate(updatedWalletInfo),
                 );
-                this.accounts.set(addressInfo.userFriendlyAddress, addressInfo);
+                this._onWalletInfoUpdate(walletInfo);
+                this.retrievalFailed = false;
+                break;
+            } catch (e) {
+                this.retrievalFailed = true;
+                if (tryCount === 5) throw e;
```

Comment:
I always like `>= 5` better, just in case the loop messes up somehow and skips 5.